legend-engine-language-pure-compiler/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/validator/MappingValidator.java

```diff
@@ -240,7 +248,54 @@ public void validateMappingElementIds(PureModel pureModel, Map<String, Mapping>
         });
     }
 
-    private void validatePropertyMappings(PureModel pureModel, org.finos.legend.pure.m3.coreinstance.meta.pure.mapping.Mapping mapping, Map<String, org.finos.legend.pure.m3.coreinstance.meta.pure.mapping.Mapping> mappingByClassMappingId)
+    private void validatePropertyMappings(PureModel pureModel, Map<String, Mapping> mappings, Map<String, org.finos.legend.pure.m3.coreinstance.meta.pure.mapping.Mapping> pureMappings)
+    {
+        pureMappings.forEach((mappingPath, mapping) ->
+        {
+                mapping._classMappings().forEach(cm ->
+                        {
+                            if (cm instanceof InstanceSetImplementation)
+                            {
+                                ((InstanceSetImplementation)cm)._propertyMappings().forEach(pm ->
+                                        {
+                                            if (pm instanceof Root_meta_pure_mapping_modelToModel_PurePropertyMapping_Impl)
+                                            {
+                                                    Root_meta_pure_mapping_modelToModel_PurePropertyMapping_Impl ppm = (Root_meta_pure_mapping_modelToModel_PurePropertyMapping_Impl) pm;
+
+                                                    if (ppm._transform instanceof Root_meta_pure_metamodel_function_LambdaFunction_Impl)
+                                                    {
+                                                        (((Root_meta_pure_metamodel_function_LambdaFunction_Impl)ppm._transform)._expressionSequence).forEach(ex ->
+                                                        {
+                                                            RichIterable<? extends FunctionExpression> feList = core_pure_corefunctions_metaExtension.Root_meta_pure_functions_meta_findExpressionsForFunctionInValueSpecification_ValueSpecification_1__Function_MANY__FunctionExpression_MANY_(
+                                                                            (ValueSpecification)ex,
+                                                                            Lists.mutable.of(pureModel.getFunction("meta::pure::functions::lang::new_Class_1__String_1__KeyExpression_MANY__T_1_",true)),
+                                                                            pureModel.getExecutionSupport());
+
+                                                            for (FunctionExpression fe: feList)
+                                                            {
+                                                                String errorMessage = "The new function/operator(^) is unsupported in property mappings, violated in property '" + ppm._transform + "' of mapping '" + mappingPath + "'";
+
+                                                                if (fe._genericType()._rawType() == ((ValueSpecification) ex)._genericType()._rawType())
```

Comment:
What if I have an expression involving new, whose return type is not the class type. Like toString(^A()). We need a better check condition here. Also,  pls write a test for such scenario.

Reply:
fe is new function call, not the complete expression. "fe._genericType()._rawType()" will be A in this case and "ex._genericType()._rawType()" will be string. 
I am adding a test for this.

Reply:
ok, makes sense